reference/enterprise-api-ref/ldap-api.markdown

```diff
@@ -0,0 +1,125 @@
+---
+layout: default
+title: LDAP authentication API
+published: true
+tags: [reference, enterprise, API, authenticating, LDAP]
+---
+
+LDAP authentication API allows to check ldap user credentials and change LDAP settings.
+
+## Login
+
+**URI:** https://hub.cfengine.com/ldap/login
+
+**Method:** POST
+
+**Parameters:**
+
+* **username** *(string)*
+    Username from LDAP
+* **password** *(string)*
+    User password
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "success": true,
+    "message": "You are successfully authenticated"
+}
+```
+
+## Get settings
+
+**URI:** https://hub.cfengine.com/ldap/settings
+
+**Method:** GET
+
+**Headers:**
+
+* **Authorization: api_token** *(string)*
+    Set token to access api
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "success": true,
+    "data": {
+        "domain_controllers": "local.loc",
+        "base_dn": "DC=local,DC=loc",
+        "login_attribute": "cn",
+        "port": 365,
+        "use_ssl": false,
+        "use_tls": false,
+        "timeout": 5,
+        "admin_username": "cn=admin,DC=local,DC=loc",
+        "admin_password": "Password is set"
```

Comment:
Literally it returns `Password is set`?

This differs from pre-ldap api settings correct?

Before the new ldap api ldap password was just stripped from the response correct?

Reply:
It returns "Password is set" to avoid showing the password.
>This differs from pre-ldap api settings correct? Before the new ldap api ldap password was just stripped from the response correct?

Yes, before the password was stripped from the response. I can remove password field from the response.

Reply:
I don't know what is more correct or desirable just asking for clarification